core/src/main/scala/org/apache/spark/h2o/backends/internal/InternalBackendUtils.scala

```diff
@@ -39,6 +41,17 @@ private[backends] trait InternalBackendUtils extends SharedBackendUtils {
       })
   }
 
+  private def resolveWorkerIPAddress(): String = {
+    val hostname = getHostname(SparkEnv.get)
+    translateHostnameToIp(hostname)
+  }
+
+  private[sparkling] def setSelfAddressToH2ONode(conf: H2OConf): Unit = {
+    if (conf.nodeNetworkMask.isEmpty) {
+      water.H2O.SELF_ADDRESS = InetAddress.getByName(resolveWorkerIPAddress())
```

Comment:
at this point the ip parameter should be in `args` https://github.com/h2oai/sparkling-water/pull/2668/files#diff-fe97e9f0f98522f054e3201152e784827ef9332e678937fdd6b139aae956db50R151

would it be more consistent if we took it from there? if the ip definition changes in getH2OWorkerArgs it would never be applied here

Reply:
np, will update, thanks!